glm/gtc/random.inl

```diff
@@ -32,8 +32,8 @@ namespace detail
 		GLM_FUNC_QUALIFIER static vec<2, uint8, P> call()
 		{
 			return vec<2, uint8, P>(
-				std::rand() % std::numeric_limits<uint8>::max(),
-				std::rand() % std::numeric_limits<uint8>::max());
+				static_cast<uint8>(std::rand()) % std::numeric_limits<uint8>::max(),
+				static_cast<uint8>(std::rand()) % std::numeric_limits<uint8>::max());
```

Comment:
I think there's something wrong here. The static cast will make the number wrap around since it will overflow, then it will use the `%` with the maximum which will do nothing. This static cast will change the result of this operation. The correct way would be this:

```c++
static_cast<uint8>(std::rand() % static_cast<int>(std::numeric_limits<uint8>::max()))
```

Then the operation would be equivalent.

Not to say that doing modulo over the rand result is okay (or even using rand at all) but a pull request that fix a build on a particular platform should at least produce the same results.